app/common/services/maps.js

```diff
@@ -95,8 +101,16 @@ function Maps(ConfigEndpoint, L, _, CONST) {
             });
         });
     }
-
+    function getSelectedBaseLayer(layer) {
+        if (!CONST.MAPBOX_API_KEY && layers.baselayers[layer].provider === 'mapbox') {
```

Comment:
I guess this could be fancier and have some `requiresKey` and mappings  , but IDK if worth it or just over complicating it. Trying to think if it'd increase readability if we did it that way. Maybe yes?